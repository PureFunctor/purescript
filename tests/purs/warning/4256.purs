-- @shouldWarnWith MissingTypeDeclaration
module Main where

addNumberSuffix :: forall a b c d. a -> b -> c -> d -> a
addNumberSuffix a _ _ _ = a

addNumberSuffix' = addNumberSuffix 0
