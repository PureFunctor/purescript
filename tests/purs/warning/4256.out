Warning found:
in module [33mMain[0m
at tests/purs/warning/4256.purs:7:1 - 7:37 (line 7, column 1 - line 7, column 37)

  No type declaration was provided for the top-level declaration of [33maddNumberSuffix'[0m.
  It is good practice to provide type declarations as a form of documentation.
  The inferred type of [33maddNumberSuffix'[0m was:
  [33m                                        [0m
  [33m  forall b6 c7 d8. b6 -> c7 -> d8 -> Int[0m
  [33m                                        [0m

in value declaration [33maddNumberSuffix'[0m

See https://github.com/purescript/documentation/blob/master/errors/MissingTypeDeclaration.md for more information,
or to contribute content related to this warning.

